_source/_posts/2022-06-28-add-auth-to-any-app-with-oauth2-proxy.adoc

```diff
@@ -0,0 +1,583 @@
+---
+layout: blog_post
+title: "Add Auth to any app with OAuth2 Proxy"
+author: brian-demers
+by: advocate
+communities: [devops,security,java]
+description: ""
+tags: [java, oauth2, oidc, oauth2-proxy]
+tweets:
+- ""
+- ""
+- ""
+image:
+type: conversion
+---
+:page-liquid:
+:toc: macro
+:experimental:
+
+Updating an application to use OAuth 2.0 doesn't need to be complicated. Most of the time, your language or framework will already have an OAuth library. Sometimes, this isn't the case, and you need to find an alternative. In this post, I'll walk through setting up and using OAuth2 Proxy to secure your application without any code changes!
+
+https://oauth2-proxy.github.io/oauth2-proxy/[OAuth2 Proxy] is a reverse proxy that sits in front of your application and handles the complexities of OpenID Connect / OAuth 2.0 for you; requests that make it to your application have already been authorized!
+
+image::{% asset_path 'blog/add-auth-to-any-app-with-oauth2-proxy/oauth2-proxy-diagram.jpg' %}[alt=OAuth2 Proxy basic usage diagram,align=center]
+
+**Prerequisites**
+
+* https://docs.docker.com/get-docker/[Docker]
```

Comment:
is docker compose pre installed on mac? on linux I have to install it as an add on